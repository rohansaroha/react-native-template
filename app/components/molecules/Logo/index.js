import React from 'react';
import { View, Image } from 'react-native';
import { images } from '@themes';

const styles = {
  logoContainer: {
    width: '100%',
    height: 100
  },
  logo: {
    width: '100%',
    height: '100%',
    alignSelf: 'center'
  }
};

function Logo() {
  return (
    <>
      <View style={styles.logoContainer}>
        <Image
          style={styles.logo}
          source={images.wednesdayLogo}
          resizeMode="contain"
        />
      </View>
    </>
  );
}

export default Logo;
