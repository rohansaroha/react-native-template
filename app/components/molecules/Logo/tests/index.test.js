/**
 *
 * Tests for Logo
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { renderWithIntl } from '@utils/testUtils';
import Logo from '../index';

describe('<Logo />', () => {
  it('Should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<Logo />);
    expect(baseElement).toMatchSnapshot();
  });
});
