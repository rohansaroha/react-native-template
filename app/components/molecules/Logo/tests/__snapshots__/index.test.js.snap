// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoWithInstructions /> Should render and match the snapshot 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "height": 150,
          "marginBottom": 25,
          "width": "100%",
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../app/assets/images/wednesday-logo.png",
          }
        }
        style={
          Object {
            "alignSelf": "center",
            "height": "100%",
            "width": "100%",
          }
        }
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "fontSize": 17,
            "fontWeight": "normal",
            "marginBottom": 5,
            "textAlign": "center",
          },
        ]
      }
      testID="t"
    >
      To get started, edit App.js
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontSize": 17,
            "fontStyle": "italic",
            "fontWeight": "normal",
            "marginBottom": 5,
            "textAlign": "center",
          },
        ]
      }
      testID="t"
    >
      
    </Text>
  </View>
</View>
`;
